core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -495,8 +495,8 @@ package object config {
     ConfigBuilder("spark.shuffle.spill.diskWriteBufferSize")
       .doc("The buffer size, in bytes, to use when writing the sorted records to an on-disk file.")
       .bytesConf(ByteUnit.BYTE)
-      .checkValue(v => v > 0 && v <= Int.MaxValue,
-        s"The buffer size must be greater than 0 and less than ${Int.MaxValue}.")
+      .checkValue(v => v > 12 && v <= Int.MaxValue,
+        s"The buffer size must be greater than 12 and less than ${Int.MaxValue}.")
```

Comment:
As recently fixed in #22705, we cannot allocate a Java array with `Int.MaxValue`. Could you please fix this, too?

Reply:
Sorry for bothering you. Can we handle this in this PR, too?

Reply:
So strange, l have handled it, but we can not see the change here

Reply:
How about using [this value](https://github.com/apache/spark/blob/master/common/unsafe/src/main/java/org/apache/spark/unsafe/array/ByteArrayMethods.java#L52)?